extension/extension.go

```diff
@@ -0,0 +1,191 @@
+	}
+	metadata := &Metadata{
+		Owner:     owner,
+		Repo:      repo,
+		Aliases:   aliases,
+		Tag:       tag,
+		LocalPath: destFilePath,
+	}
+	e.manifest.Metadatas = append(e.manifest.Metadatas, metadata)
+	e.manifest.Update = time.Now()
+	return FlushManifest(e.manifest)
+}
+
+func (e *Extension) downloadAsset(url, destPath string) error {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return fmt.Errorf("error creating request: %v", err)
+	}
+	req.Header.Set("Accept", "application/octet-stream")
+
+	resp, err := e.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error executing request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode >= 300 {
+		return e.getResponseError(resp)
+	}
+
+	f, err := os.OpenFile(destPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0755)
+	if err != nil {
+		return fmt.Errorf("error opening file: %v", err)
+	}
+	defer f.Close()
+
+	_, err = io.Copy(f, resp.Body)
+	if err != nil {
+		return fmt.Errorf("error copying file: %v", err)
+	}
+	return nil
+}
+
+func (e *Extension) getResponseError(resp *http.Response) error {
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("error reading body: %v", err)
+	}
+	var buff bytes.Buffer
+	if err := json.Indent(&buff, body, "", "  "); err != nil {
+		return fmt.Errorf("error indenting json: %v", err)
+	}
+	return errors.New(string(buff.Bytes()))
+}
+
+func (e *Extension) getDownloadURL(release *github.RepositoryRelease) (string, error) {
+	currentDist := e.getCurrentDist()
+	for _, asset := range release.Assets {
+		if asset.Name != nil {
+			if strings.HasSuffix(*asset.Name, currentDist) && asset.BrowserDownloadURL != nil {
+				return asset.GetBrowserDownloadURL(), nil
+			}
+		}
+	}
+	return "", fmt.Errorf("asset for [%s] is not found", currentDist)
+}
+
+func (e *Extension) getCurrentDist() string {
+	return runtime.GOOS + "-" + runtime.GOARCH
+}
+
+func (e *Extension) validateInstall(owner, repo, alias string) error {
+	if !strings.HasPrefix(repo, defaultRepoPrefix) {
+		return fmt.Errorf("[%s] does not have prefix [%s]", repo, defaultRepoPrefix)
+	}
+	name := strings.TrimPrefix(repo, defaultRepoPrefix)
+	for _, c := range e.reservedCommands {
+		if name == c {
+			return fmt.Errorf("[%s] is reserved command", name)
+		}
+	}
+	for _, metadata := range e.manifest.Metadatas {
+		if owner == metadata.Owner && repo == metadata.Repo {
+			return fmt.Errorf("%s/%s [%s] is already installed", owner, repo, metadata.Tag)
+		}
+		for _, a := range metadata.Aliases {
+			if alias == a {
+				return fmt.Errorf("alias [%s] is already used", alias)
+			}
+		}
+	}
+	return nil
+}
+
+// Run executes extension
+func (e *Extension) Run(name string, args []string) error {
+	var path string
+	for _, metadata := range e.manifest.Metadatas {
+		for _, a := range metadata.Aliases {
+			if name == a {
+				path = metadata.LocalPath
+			}
+		}
+	}
+	cmd := exec.Command(path, args...)
+	cmd.Stdout = os.Stdout
```

Comment:
I think we should also pipe `Stdin` and `Stderr`

Reply:
Okay, will add it later along with the others.